cmd/podman/system_df.go

```diff
@@ -0,0 +1,363 @@
+		Type:        "Images",
+		TotalCount:  numberOfImages,
+		Active:      numberOfActiveImages,
+		Size:        sumSizeStr,
+		Reclaimable: reclaimableStr,
+	}
+	return imageDiskUsage, nil
+}
+
+func imageUsedSize(img *image.Image, imgUniqueSizeMap map[string]uint64, imageUsedbyCintainerMap map[string][]*libpod.Container, imageUsedbyActiveContainerMap map[string][]*libpod.Container) uint64 {
+	var usedSize uint64
+	imgUnique := imgUniqueSizeMap[img.ID()]
+	if _, isCtrActive := imageUsedbyActiveContainerMap[img.ID()]; isCtrActive {
+		usedSize += imgUnique
+	} else {
+		containers := imageUsedbyCintainerMap[img.ID()]
+		for _, ctr := range containers {
+			if len(ctr.UserVolumes()) > 0 {
+				usedSize += imgUnique
+				return usedSize
+			}
+		}
+	}
+	return usedSize
+}
+
+func imagesUsedbyCtr(containers []*libpod.Container, activeContainers map[string]*libpod.Container) (map[string][]*libpod.Container, map[string][]*libpod.Container, error) {
+	imgCtrMap := make(map[string][]*libpod.Container)
+	imgActiveCtrMap := make(map[string][]*libpod.Container)
+	for _, ctr := range containers {
+		imgID, _ := ctr.Image()
+		imgCtrMap[imgID] = append(imgCtrMap[imgID], ctr)
+		if _, isActive := activeContainers[ctr.ID()]; isActive {
+			imgActiveCtrMap[imgID] = append(imgActiveCtrMap[imgID], ctr)
+		}
+	}
+	return imgCtrMap, imgActiveCtrMap, nil
+}
+
+func getContainerDiskUsage(containers []*libpod.Container, activeContainers map[string]*libpod.Container) (systemDfDiskUsage, error) {
+	var (
+		sumSize        int64
+		unreclaimable  int64
+		reclaimableStr string
+	)
+	for _, ctr := range containers {
+		size, err := ctr.RWSize()
+		if err != nil {
+			return systemDfDiskUsage{}, errors.Wrapf(err, "error getting size of container %s", ctr.ID())
+		}
+		sumSize += size
+	}
+	for _, activeCtr := range activeContainers {
+		size, err := activeCtr.RWSize()
+		if err != nil {
+			return systemDfDiskUsage{}, errors.Wrapf(err, "error getting size of active container %s", activeCtr.ID())
+		}
+		unreclaimable += size
+	}
+	if sumSize == 0 {
+		reclaimableStr = fmt.Sprintf("%s (%v%%)", units.HumanSizeWithPrecision(0, 3), 0)
+	} else {
+		reclaimable := sumSize - unreclaimable
+		reclaimableStr = fmt.Sprintf("%s (%v%%)", units.HumanSizeWithPrecision(float64(reclaimable), 3), 100*reclaimable/sumSize)
+	}
+	containerDiskUsage := systemDfDiskUsage{
+		Type:        "Containers",
+		TotalCount:  len(containers),
+		Active:      len(activeContainers),
+		Size:        units.HumanSizeWithPrecision(float64(sumSize), 3),
+		Reclaimable: reclaimableStr,
+	}
+	return containerDiskUsage, nil
+}
+
+func ctrIsActive(ctr *libpod.Container) (bool, error) {
+	state, err := ctr.State()
+	if err != nil {
+		return false, err
+	}
+	return state == libpod.ContainerStatePaused || state == libpod.ContainerStateRunning, nil
+}
+
+func activeContainers(runtime *libpod.Runtime) (map[string]*libpod.Container, error) {
+	containers, err := runtime.GetAllContainers()
```

Comment:
Call this once and pass it into the function calls.  You are calling this many times. There is a potential that it could change and cause you problems.